app/models/topic_user.rb

```diff
@@ -40,7 +40,7 @@ def notification_reasons
     def auto_notification(user_id, topic_id, reason, notification_level)
       should_change = TopicUser
         .where(user_id: user_id, topic_id: topic_id)
-        .where("notifications_reason_id IS NULL OR (notification_level < :min AND notification_level > :max)", min: notification_level, max: notification_levels[:regular])
+        .where("notifications_reason_id IS NULL OR (notification_level < :max AND notification_level > :min)", max: notification_level, min: notification_levels[:regular])
```

Comment:
Is this change intentional because the code path that has been updated doesn't seem to be affected by this change here.

Reply:
Yeah it's not related but I was reading the code and found the placeholders names very confusing. It's just mathematically impossible to find a value larger than `max` and smaller than `min` at the same time.

Reply:
@OsamaSayegh I wonder if this might be a on-demand fix for a bug that was introduced in the past. Maybe have a look through the git commit history?